### PR TITLE
Added changeFullname Resolver

### LINKED ISSUE #72: `changeFullName` user Mutation
Currently there is no way for a registered user to change its `fullName`. Create a `changeFullName` mutation which will let users change their current `fullName`.

### PR DESCRIPTION
**Pull request by hazardemircan**
- changeFullnameResolver added
- Tests will be committed

### CODE REVIEW

**Comment by d-e-v-esh** [comment]
@hazardemircan Hey, I didn't run your code yet, I just looked through it. I think I kind of understand what you are trying to do. But can you still explain a little bit why we need `loginInfo` to change the `fullName` so that there is no confusion?

**Comment by hazardemircan** [comment]
> @hazardemircan Hey, I didn't run your code yet, I just looked through it. I think I kind of understand what you are trying to do. But can you still explain a little bit why we you `loginInfo` to change the `fullName` so that there is no confusion?

I couldn't get the `userId ` from  the `session` , that is why.

**Comment by d-e-v-esh** [comment]
@hazardemircan You should be able to definitely. Did you install Redis and make sure it's up and running?

**Comment by hazardemircan** [comment]
> @hazardemircan You should be able to definitely. Did you install Redis and make sure it's up and running?

I'll try again now, probably problem was Redis, thanks

File: src/resolvers/userResolver.js [hunk 1]
```diff
@@ -145,6 +145,62 @@ const userResolver = {
...
+            {
+              field: "password",
+              message: "incorrect password",
+            },
+          ],
+        };
+      }
+
+      return { user };
+    },
```
Comment (JPTIZ) [inline src/resolvers/userResolver.js:203]: (Just passing!)

Regardless of if this is accepted, small coding tip: early-return :) First check for possible failures, then do what you need with the data knowing the (known) issues have been all handled.

```suggestion
      if (!isPasswordValid) {
        return {
          errors: [
            {
              field: "password",
              message: "incorrect password",
            },
          ],
        };
      }
      
      if (user.fullName === newFullname) {
        return {
          errors: [
            {
              field: "newFullname",
              message: "newFullname cannot be the same as the fullName",
            },
          ],
        };
      }
      
      await user.updateOne({
        fullName: newFullname,
        updatedAt: new Date(),
      });

      return await User.findOne();
    },
```

**Comment by hazardemircan** [comment]
> @hazardemircan You should be able to definitely. Did you install Redis and make sure it's up and running?

@d-e-v-esh Hey i just checked the session and after login or register; session do not contain userId. Am i missing something, i would like to get your help. If i'm not missing something
Additionally logout mutation don't work If i'm not missing something, you may want to check.

Thanks...

**1. Login Mutation**

Login Mutation


**2. redis-cli** 

redis-cli

**3. changeFullName Mutation** 

changeFullName Mutation

File: src/resolvers/userResolver.js (hunk 1 above)
Comment (hazardemircan) [reply src/resolvers/userResolver.js:203]: Thanks for advice, i understood the principle.

Comment (hazardemircan) [reply src/resolvers/userResolver.js:203]: > Thanks for suggestion, i understood the principle.

**Comment by d-e-v-esh** [comment]
image

@hazardemircan Isn't this the `userId`?

**Comment by hazardemircan** [comment]
> image
> 
> @hazardemircan Isn't this the `userId`?

this was from login mutation, below section when i `console.log(req.session.userId)` in `changeFullname` mutation

**Comment by hazardemircan** [comment]
> > image
> > @hazardemircan Isn't this the `userId`?
> 
> this was from login mutation, below section when i `console.log(req.session.userId)` in `changeFullname` mutation

@d-e-v-esh i can't get `userId `in `changeFullname`

**Comment by d-e-v-esh** [comment]
Ohh, so `userId` is not available in `changeFullName`?

**Comment by d-e-v-esh** [comment]
I see

**Comment by hazardemircan** [comment]
> Ohh, so `userId` is not available in `changeFullName`?

Also logout doesn't work maybe because of this

**Comment by d-e-v-esh** [comment]
@hazardemircan  I know why the `logout` mutation isn't working. You need to set up Apollo Server cookies. For that you need to go to settings in Apollo Studio, turn on “Include Cookies” and add a new shared header with header key as `x-forwarded-proto` and header value as `https`.

**Comment by d-e-v-esh** [comment]
image

**Comment by d-e-v-esh** [comment]
@hazardemircan 
You should be easily able to execute the `login` and `logout` Mutation.

brave_1K6bHkduHi

**Comment by d-e-v-esh** [comment]
> @d-e-v-esh i can't get `userId `in `changeFullname`

@hazardemircan You didn't import `req` at line 149 in the `userResolver` file. That's why you can't access `userId`. You can see how it is imported in other Mutations.

**Comment by hazardemircan** [comment]
@d-e-v-esh oh, i didn't knew that. I will commit today for sure.

**Comment by d-e-v-esh** [comment]
Yaa, I also didn't think to put all this in the docs. I will make a list of all these things and put it in the README.md file so that other people won't have this problem 👍

**Pull request closed without merging**